mailpoet/lib/Cron/Workers/SendingQueue/SendingQueue.php

```diff
@@ -274,6 +274,8 @@ private function processSending(SendingTask $queue, int $timer): void {
         if (!$isTransactional) {
           $now = Carbon::createFromTimestamp((int)current_time('timestamp'));
           $this->subscribersRepository->bulkUpdateLastSendingAt($foundSubscribersIds, $now);
+          // We're nullifying this value so these subscribers' engagement score will be recalculated the next time the cron runs
+          $this->subscribersRepository->bulkUpdateEngagementScoreUpdatedAt($foundSubscribersIds, null);
```

Comment:
I did think it would be nice to execute both of these updates in the same query, but I wanted to keep the methods separate in case we ever wanted to use them separately in other contexts, and to keep them easier/clearer to test.

My guess was that this wouldn't cause much of a performance hit, but it's probably worthwhile to do everything we can to minimize the query count during the sending process.

I decided to wrap the updates in a transaction in 0783c6ec0db9e47298f503957ff7b682b6ee3261, which should address this issue for anyone using InnoDB. Does this seem like a reasonable approach?

Reply:
@johnolek My knowledge of InnoDB isn't very profound. So just to confirm I understand. Using the transaction still means DB needs to run two queries, but the expected performance gain comes from the data being written only once. Correct?

Reply:
@costasovo All I was thinking was that since InnoDB supports transactions and MyISAM doesn't, if a user is using InnoDB this might result in only a single round trip to the database. I think I may have been mistaken though, and using a transaction doesn't necessarily mean the operations would only require a single round trip.

Do you think it would be worthwhile to create a more generic bulk update method, one that accepts key/value pairs and can bulk update multiple columns in a single query? 

My suspicion is that the code should be ok as is -- processing a sending queue already requires lots of separate DB queries and this one shouldn't be very slow. I might ask if QA can do a before/after test to see if this change affects sending time for very large sends. We can always optimize this later if we discover it's causing performance issues.